test/jdk/javax/swing/JPopupMenu/6694823/bug6694823.java

```diff
@@ -21,20 +21,27 @@
  * questions.
  */
 
+
+import java.awt.Dimension;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import javax.swing.JFrame;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.JPopupMenu;
+import javax.swing.SwingUtilities;
+
 /*
  * @test
  * @key headful
  * @bug 6694823
  * @summary Checks that popup menu cannot be partially hidden
- * by the task bar in applets.
- * @author Mikhail Lapshin
- * @run main/othervm -Djava.security.manager=allow bug6694823
+ *          by the task bar.
```

Comment:
This particular test was failing on windows & linux after SM removal. There is a functional issue and for that reason I think it is better to retain this test. Details documented here - [JDK-8342012](https://bugs.openjdk.org/browse/JDK-8342012)

Reply:
The updated test `bug6694823.java` works correctly on Windows and displays its popup over the Windows taskbar — it is expected.

The popup had to be moved if the security manager didn't allow to call `setAlwaysOnTop(true)`.

> There is a functional issue and for that reason I think it is better to retain this test. Details documented here - [JDK-8342012](https://bugs.openjdk.org/browse/JDK-8342012)

There's no functional issue. The `bug6694823.java` test was designed to pass **with the security manager**.

The `bug6694823.java` test fails without the security manager because the conditions don't hold any more.

Reply:
@aivanov-jdk 
> The updated test bug6694823.java works correctly on Windows and displays its popup over the Windows taskbar — it is expected.

On macOS, popup is shifted up and does not cover the taskbar even without SM. 
On native applications (eg. Word) popup menu don't overlap taskbar when clicked close to taskbar so do we consider this as expected behavior or the way it works currently (overlaps the taskbar)?

Reply:
The pop is shifted up on macOS because `LWCToolkit` returns `false` in `canPopupOverlapTaskBar`:

https://github.com/openjdk/jdk/blob/36d71735e3554264e8d17f7e0e72999ac639e398/src/java.desktop/macosx/classes/sun/lwawt/macosx/LWCToolkit.java#L900-L902

On Windows, apps can control the area where they want a popmenu display or exclude an area of the screen. At the same time, Firefox displays its right-click menu over the taskbar if the menu fits when dropped down. A native Win32 app with a menu bar displays popup menus over the taskbar if it fits on the screen; some menus could open upwards if their items don't fit on the screen to display downwards (I attached a screenshot to JBS). Popup menus in Win32 Notepad (the version in Windows 10) displays it's right-click menu over the taskbar if it fits.

At the same time, right-click menu of a window title never displays over the taskbar, the menu opens upward if it doesn't fit without covering part of the taskbar.

I'm pretty sure the default return value of `true` from `SunToolkit.canPopupOverlapTaskBar` isn't something that was chosen accidentally.

Windows applications may avoid showing popup over the taskbar (confining the popups inside the work area, see [`SystemParametersInfoW`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-systemparametersinfow) and `SPI_GETWORKAREA`). We can change the behaviour of the JDK.

Yet the failure of `bug6694823.java` without the security manager isn't a product bug in my opinion.

Reply:
Fixed in https://github.com/openjdk/jdk/pull/21498/commits/aca9555a0b697bd9829224396a5448c88057009d